### PR TITLE
[relay-runtime] Add types for RelayModernEnvironment

### PR DESCRIPTION
**Pull request by robrichard**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/facebook/relay/blob/v10.1.2/packages/relay-runtime/store/RelayStoreTypes.js#L672-L692
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## relay-runtime ([<kbd>unpkg</kbd>](https://unpkg.com/browse/relay-runtime@latest/))
was missing the following properties:
1. ReplaySubject
1. getNode
1. reportMissingRequiredFields
1. isFragment
1. isInlineDataFragment

<details>
<summary>as well as these 13 other properties...</summary>
<p>isRequest, MutationTypes, RangeOperations, MutationHandlers, VIEWER_ID, VIEWER_TYPE, PreloadableQueryRegistry, createPayloadFor3DField, RelayError, DEFAULT_HANDLE_KEY, isScalarAndEqual, recycleNodesInto, stableCopy</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 75a6b919cb489e377e5f8c74b6f5f8d15ffb35e6
</p>

File: types/relay-runtime/lib/store/RelayModernEnvironment.d.ts [hunk 1]
```diff
@@ -41,6 +41,7 @@ export interface EnvironmentConfig {
 export default class RelayModernEnvironment implements Environment {
     options: unknown;
     configName: string | null | undefined;
+    requiredFieldLogger: RequiredFieldLogger;
```
Comment (Stephen2) [inline types/relay-runtime/lib/store/RelayModernEnvironment.d.ts:44]: Can you clarify Flow source code for this line?  I checked:
https://github.com/facebook/relay/blob/ba9a3f031f8adc0f60bef603a76385433518cd90/packages/relay-runtime/store/RelayModernEnvironment.js

And couldn't find it

Comment (robrichard) [reply types/relay-runtime/lib/store/RelayModernEnvironment.d.ts:44]: Without it I get these errors:

```
Property 'requiredFieldLogger' is missing in type 'RelayModernEnvironment' but required in type 'Environment'.
    Type 'RelayModernEnvironment' is not assignable to type 'Environment'.

  node_modules/@types/relay-runtime/lib/store/RelayStoreTypes.d.ts:660:3
    660   requiredFieldLogger: RequiredFieldLogger;
          ~~~~~~~~~~~~~~~~~~~
    'requiredFieldLogger' is declared here.
  node_modules/@types/react-relay/index.d.ts:107:5
    107     environment: IEnvironment;
            ~~~~~~~~~~~
    The expected type comes from property 'environment' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<ReactRelayQueryRenderer<AnchorListTableQuery>> & Readonly<...> & Readonly<...>'
  node_modules/@types/react-relay/index.d.ts:107:5
    107     environment: IEnvironment;
            ~~~~~~~~~~~
    The expected type comes from property 'environment' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<ReactRelayQueryRenderer<AnchorListTableQuery>> & Readonly<...> & Readonly<...>'

```

Comment (Stephen2) [reply types/relay-runtime/lib/store/RelayModernEnvironment.d.ts:44]: Thanks. Found it.  I was looking in the wrong git hash in the Flow types.

Found here:
https://github.com/facebook/relay/blob/v10.1.2/packages/relay-runtime/store/RelayModernEnvironment.js

**Review by Stephen2 — APPROVED** [review]
:+1:

**Pull request merged**